src/Development/IDE/Core/Shake.hs

```diff
@@ -567,16 +567,22 @@ shakeRestart IdeState{..} acts =
               let profile = case res of
                       Just fp -> ", profile saved at " <> fp
                       _ -> ""
-              logDebug (logger shakeExtras) $ T.pack $
-                  "Restarting build session (aborting the previous one took " ++
-                  showDuration stopTime ++ profile ++ ")"
+              let msg = T.pack $ "Restarting build session (aborting the previous one took "
+                              ++ showDuration stopTime ++ profile ++ ")"
+              logDebug (logger shakeExtras) msg
+              eventer shakeExtras (notifyLogMessage msg)
```

Comment:
we should probably only send this notification when `--test`

If the goal is to observe session restarts, shouldn't this notification be sent from `newSession`?

https://github.com/haskell/ghcide/blob/20ce9d3a46e92b57ab3e559d20d8f6b4686c8bfa/src/Development/IDE/Core/Shake.hs#L605

Reply:
Mmm,  i have followed closely the existing log messages, although i did not for "finishing" ones.
The "Restarting build session..." message seemed the right one to know if, well, the session was being restarted. 🤔

Reply:
ignore me, the notification is indeed in the right place